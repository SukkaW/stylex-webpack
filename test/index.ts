import { describe, it } from 'mocha';
import getWebpackCompiler from './utils/get-webpack-compiler';

import compile from './utils/compile';
import getModuleSource from './utils/get-module-source';
import { jestExpect as expect } from 'mocha-expect-snapshot';

([
  ['stylex-webpack (webpack)', getWebpackCompiler]
  // ['forgetti-loader (rspack)', getRspackCompiler]
] as const).forEach(([name, getCompiler]) => {
  describe(name, () => {
    it('should work', async () => {
      const [compiler, fs] = getCompiler('./simple.js');
      const stats = await compile(compiler);

      // const { chunks, modules } = stats.toJson({ source: true });

      expect(getModuleSource('./simple.js', stats, fs)).toMatchSnapshot('simple.js');
    });
  });
});
