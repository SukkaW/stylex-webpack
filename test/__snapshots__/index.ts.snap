// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`stylex-webpack (webpack) should work: simple.js 1`] = `
"import * as stylex from '@stylexjs/stylex';
const fadeAnimation = "xgnty7z-B";
export default function App() {
  return {
    className: "xeuoslp x78zum5 x1hm9lzh xlrshdv x1egiwwb x1oz5o6v"
  };
}"
`;
